src/actions/mod.rs

```diff
@@ -532,10 +535,11 @@ impl InitActionContext {
                                                   file: &Path,
                                                   out: &O) {
         if !self.config.lock().unwrap().linting_enabled {
+            self.pending_direct_results.store(true, Ordering::SeqCst);
```

Comment:
The functionality of this flag confuses me.
- The only time the flag is different from the notifier being active is in the case on this line, but why is the flag actually set here?
- The flag is also never _unset_. But will remain true forever after either signaling progress start once or when triggering lint analysis once

Reply:
This flag **is** unset once the lint_analysis is done `src/server/mod.rs:418`.
We can rename this, because its still referencing the "direct_only" mode, which is now going to be cli_mode.

When the server is used with VSCode or an IDE client, this flag should not have any effect. We added this only to make sure the DFA client does not receive as "progress_end" before the lint_analysis' diagnostics are done and notified to the client.

The scenario you are pointing out here does seem strange, it might be unnecessary to set this flag to true in this case, given linting_enabled is not even set in this case. We will revisit this and update this part of the logic, as well as rename the flag to better reflect the purpose its serving.

Reply:
Yes, the intention was actually to set this flag to false in this case. Already correcting this locally.
This means if linting is disabled, the server should no longer wait for pending results before issuing "progress_end" to the client.

Reply:
We are now setting this to false at this point.
That is, its not required for the server to wait for linter to be finished before sending progress_end because linter is not even enabled in this case. In other words, there are no pending results after Isolated or Device analyses are done.

We renamed the flag, hope this makes its functionality clearer now.